TCAT/Controllers/OptionsViewController.swift

```diff
@@ -196,48 +184,91 @@ class OptionsViewController: UIViewController, UITableViewDelegate, UITableViewD
     //MARK: Search bar functionality
     func searchForRoutes(){
         //N2SELF: N2 put in parameters for searchDate & searchDepature
-        if let startBus = searchFrom.1 as? BusStop, let end = searchTo.1 {
-            switch searchTo.0 {
-            case .busstop:
-                if let endBus = end as? BusStop{
-                    routes = loaderroutes
-                    Loader.addLoaderTo(routeResults)
-                    routeResults.reloadData()
-                    Network.getBusRoute(startLat: startBus.lat!, startLng: startBus.long!, destLat: endBus.lat!, destLng: endBus.long!).perform(withSuccess: { (routes) in
-                        print("success")
-//                        routes.map({ route in
-//                            route.directions.map({ (direction) in
-//                                <#code#>
-//                            })
-//                        })
-                        self.routes = routes
-                        self.routeResults.reloadData()
-                        self.loaded()
-                    }, failure: { (error) in
-                        print("Error: \(error)")
-                        self.routes = []
-                        self.routeResults.reloadData()
-                        self.loaded()
-                    })
-                }
-            default: //place result
-                if let endPlace = end as? PlaceResult{
-                    Loader.addLoaderTo(routeResults)
-                    routes = loaderroutes
-                    routeResults.reloadData()
-                    Network.getPlaceRoute(startLat: startBus.lat!, startLng: startBus.long!, destPlaceID: endPlace.placeID!).perform(withSuccess: { (routes) in
-                        self.routes = routes
-                        self.routeResults.reloadData()
-                        self.loaded()
-                    }, failure: { (error) in
-                        print("Error: \(error)")
-                        self.routes = []
-                        self.routeResults.reloadData()
-                        self.loaded()
-                    })
-                }
-            }
+        if searchTime == nil{
+            searchTime = Date()
+        }
+        
+        let (fromBus, fromPlace) = searchFrom
+        let (toBus, toPlace) = searchTo
+        if let startBus = fromBus, let endBus = toBus{
+            Network.getRoutes(start: startBus, end: endBus, time: searchTime!, type: searchTimeType).perform(withSuccess: { (routes) in
+                self.routes = self.getValidRoutes(routes: routes)
+                self.routeResults.reloadData()
+                self.loaded()
```

Comment:
You do the same thing in both blocks, can you put the duplicate code after the failure brackets?